src/main/scala/org/broadinstitute/dsde/agora/server/business/PermissionBusiness.scala

```diff
@@ -1,105 +1,167 @@
+    permissionsDataSource.inTransaction { db =>
+      authEntityRequester(db, entity, requester, accessObject) {
+        db.aePerms.editEntityPermission(entity, accessObject)
+      }
+    }
   }
 
-  def deleteEntityPermission(entity: AgoraEntity, requester: String, userToRemove: String): Int = {
-    checkSameRequester(requester, userToRemove)
-    authorizeEntityRequester(entity, requester)
-    AgoraEntityPermissionsClient.deleteEntityPermission(entity, userToRemove)
+  def deleteEntityPermission(entity: AgoraEntity, requester: String, userToRemove: String): Future[Int] = {
+    Try(checkSameRequester(requester, userToRemove)) match {
+      case Failure(regret) => Future.failed(regret)
+      case Success(_) => {
+        permissionsDataSource.inTransaction { db =>
+          authEntityRequester(db, entity, requester) {
+            db.aePerms.deleteEntityPermission(entity, userToRemove)
+          }
+        }
+      }
+    }
   }
 
 
   /**
     * Utility method to both authorize the namespace requester and check that the requester is not modifying their own
     * permissions
     */
-  private def authorizeNamespaceRequester(entity: AgoraEntity, requester: String, accessObject: AccessControl): Unit = {
-    val namespaceACLs = getNamespaceACLs(entity, requester)
-    checkSameRequesterAndPermissions(namespaceACLs.find(acl => acl.user.equals(requester)), accessObject)
-    if (!namespaceACLs.exists(_.roles.canManage))
-      throw NamespaceAuthorizationException(AgoraPermissions(Manage), entity, requester)
+  private def authNamespaceRequester[T](db: DataAccess, entity: AgoraEntity, requester: String, accessObject: AccessControl)(op: => ReadWriteAction[T]): ReadWriteAction[T] = {
+    withNamespaceACLs(db, entity, requester) { namespaceACLs =>
+      checkSameRequesterAndPermissions(namespaceACLs.find(acl => acl.user.equals(requester)), accessObject)
+      authorizeNamespaceRequester(db, namespaceACLs, entity, requester) flatMap { _ => op }
+    }
   }
 
-  private def authorizeNamespaceRequester(entity: AgoraEntity, requester: String): Unit = {
-    authorizeNamespaceRequester(getNamespaceACLs(entity, requester), entity, requester)
+  private def authNamespaceRequester[T](db: DataAccess, entity: AgoraEntity, requester: String)(op: => ReadWriteAction[T]): ReadWriteAction[T] = {
+    withNamespaceACLs(db, entity, requester) { namespaceACLs =>
+      authorizeNamespaceRequester(db, namespaceACLs, entity, requester) flatMap { _ => op }
+    }
+
   }
 
-  private def authorizeNamespaceRequester(acls: Seq[AccessControl], entity: AgoraEntity, requester: String): Unit = {
-    if (!acls.exists(_.roles.canManage))
-      throw NamespaceAuthorizationException(AgoraPermissions(Manage), entity, requester)
+  private def authorizeNamespaceRequester(db: DataAccess, acls: Seq[AccessControl], entity: AgoraEntity, requester: String): ReadWriteAction[Unit] = {
+    //NOTE: The use of aePerms here seems wrong, but it's not!
```

Comment:
Can you explain this a little more? First, it didn't jump out at me as looking wrong, but that may just be my unfamiliarity with this code. Second, if it did jump out at me, I'd want the comment to say more about why it's not wrong. (Ideally, the code could be changed so that it doesn't seem wrong.)